corehq/apps/settings/forms.py

```diff
@@ -294,10 +304,15 @@ def create_key(self, user):
             HQApiKey.objects.get(name=self.cleaned_data['name'], user=user)
             raise DuplicateApiKeyName
         except HQApiKey.DoesNotExist:
+            if self.cleaned_data['domain'] and self.cleaned_data['domain'] != self.ALL_DOMAINS:
+                domain = self.cleaned_data['domain']
+            else:
+                domain = ''
             new_key = HQApiKey.objects.create(
                 name=self.cleaned_data['name'],
                 ip_allowlist=self.cleaned_data['ip_allowlist'],
                 user=user,
+                domain=domain,
```

Comment:
Nit: if line 266 was changed to `ALL_DOMAINS = ''` could this be changed to
```suggestion
            new_key = HQApiKey.objects.create(
                name=self.cleaned_data['name'],
                ip_allowlist=self.cleaned_data['ip_allowlist'],
                user=user,
                domain=self.cleaned_data['domain'] or '',
```

Reply:
good call https://github.com/dimagi/commcare-hq/pull/28493/commits/09281b09291ae0885d17e662fde776177d1559e7